tools/clang/test/HLSLFileCheck/hlsl/intrinsics/basic/fmod_const_eval.hlsl

```diff
@@ -0,0 +1,55 @@
+// RUN: %dxc -T vs_6_0 %s -E main | %FileCheck %s
+// RUN: %dxc -T vs_6_0 %s -E main -DNO_FOLD | %FileCheck %s -check-prefixes=NO_FOLD
+
+// Ensure fmod is constant evaluated during codegen, or dxil const eval
+// TODO: handle fp specials properly!
+
+RWBuffer<float4> results : register(u0);
+
+[shader("vertex")]
+void main(bool b : B) {
+    uint i = 0;
+
+    // Literal float
+    // 2.5, -2.5, 2.5, -2.5
+    // CHECK: call void @dx.op.bufferStore.f32(i32 69, %dx.types.Handle %{{.+}}, i32 0, i32 undef, float 2.500000e+00, float -2.500000e+00, float 2.500000e+00, float -2.500000e+00, i8 15)
+    results[i++] = float4(fmod(5.5, 3.0),
+                          fmod(-5.5, 3.0),
+                          fmod(5.5, -3.0),
+                          fmod(-5.5, -3.0));
+
+    // Explicit float
+    // 2.5, -2.5, 2.5, -2.5
+    // CHECK: call void @dx.op.bufferStore.f32(i32 69, %dx.types.Handle %{{.+}}, i32 1, i32 undef, float 2.500000e+00, float -2.500000e+00, float 2.500000e+00, float -2.500000e+00, i8 15)
+    results[i++] = float4(fmod(5.5f, 3.0f),
+                          fmod(-5.5f, 3.0f),
+                          fmod(5.5f, -3.0f),
+                          fmod(-5.5f, -3.0f));
+
+#ifdef SPECIALS
+    // Literal float
+    // 0.0, -0.0, NaN, -NaN
```

Comment:
I realize this isn't run yet, but could there be CHECKs for when it is?

Reply:
I'm not positive these are the expected values yet for `-Gis`.  I think there are also more cases to add for FP specials.  I had left this in because I didn't want to lose track of the scenario so far, but maybe it's better to just delete this code and have an issue tracking the work to handle and test FP specials in constant evaluation later in a different solution merged to main.  What do you think?